### ISSUE TITLE
[artifactory-oss] Failed to parse user system yaml - could not find expected ':'

### ISSUE THREAD

**Issue by rblaine95** [jfrog/charts#1630]
**Is this a request for help?**: Yes

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): Possibly bug?

**Version of Helm and Kubernetes**:
* Helm: `version.BuildInfo{Version:"v3.9.0", GitCommit:"7ceeda6c585217a19a1131663d8cd1f7d641b2a7", GitTreeState:"clean", GoVersion:"go1.18.2"}`
* Kubernetes: `Server Version: v1.21.9-eks-0d102a7`

**Which chart**: Artifactory OSS

**What happened**:
After upgrading `107.37.14` -> `107.38.10`, the Artifactory Frontend fails to start up:
<img width="1421" alt="image" src="https://user-images.githubusercontent.com/4052340/169500571-2b3f87f3-6b71-4aee-a2b7-12e55ce10842.png">

The following error is logged:
```
» k logs -f artifactory-artifactory-0
Preparing to run Artifactory in Docker
Running as uid=1030(artifactory) gid=1030(artifactory) groups=1030(artifactory)
Dockerfile for this image can found inside the container.
To view the Dockerfile: 'cat /docker/artifactory-oss/Dockerfile.artifactory'.
Resolved .shared.database.type (postgresql) from /opt/jfrog/artifactory/var/etc/system.yaml
Resolved JF_SHARED_DATABASE_URL (jdbc:postgresql://xxxxxx.cluster-xxxxxx.eu-west-1.rds.amazonaws.com:5432/artifactory) from environment variable
Waiting for DB postgresql to be ready on xxxxxx.cluster-xxxxxx.eu-west-1.rds.amazonaws.com/5432 for 30 seconds
Copying Artifactory bootstrap files
2022-05-20T09:07:34.404Z [shell] [INFO ] [] [systemYamlHelper.sh:605       ] [main] - Resolved .artifactory.tomcat.connector.maxThreads (200) from /opt/jfrog/artifactory/var/etc/system.yaml
2022-05-20T09:07:34.546Z [shell] [INFO ] [] [systemYamlHelper.sh:605       ] [main] - Resolved .access.tomcat.connector.maxThreads (50) from /opt/jfrog/artifactory/var/etc/system.yaml
2022-05-20T09:07:34.681Z [shell] [INFO ] [] [systemYamlHelper.sh:605       ] [main] - Resolved .artifactory.tomcat.connector.extraConfig (acceptCount="100") from /opt/jfrog/artifactory/var/etc/system.yaml
2022-05-20T09:07:34.804Z [shell] [INFO ] [] [systemYamlHelper.sh:605       ] [main] - Resolved .access.tomcat.connector.extraConfig (acceptCount="100") from /opt/jfrog/artifactory/var/etc/system.yaml
2022-05-20T09:07:35.022Z [shell] [INFO ] [] [systemYamlHelper.sh:605       ] [main] - Resolved .shared.extraJavaOpts (__sensitive_key_hidden___) from /opt/jfrog/artifactory/var/etc/system.yaml
2022-05-20T09:07:35.146Z [shell] [INFO ] [] [systemYamlHelper.sh:605       ] [main] - Resolved .shared.extraJavaOpts (__sensitive_key_hidden___) from /opt/jfrog/artifactory/var/etc/system.yaml
2022-05-20T09:07:35.678Z [shell] [INFO ] [] [systemYamlHelper.sh:605       ] [main] - Resolved .shared.database.type (postgresql) from /opt/jfrog/artifactory/var/etc/system.yaml
2022-05-20T09:07:35.940Z [shell] [INFO ] [] [systemYamlHelper.sh:605       ] [main] - Resolved JF_SHARED_DATABASE_URL (jdbc:postgresql://xxxxxx.cluster-xxxxxx.eu-west-1.rds.amazonaws.com:5432/artifactory) from environment variable
2022-05-20T09:07:36.100Z [shell] [INFO ] [] [systemYamlHelper.sh:605       ] [main] - Resolved JF_SHARED_DATABASE_PASSWORD (__sensitive_key_hidden___) from environment variable
2022-05-20T09:07:36.240Z [shell] [INFO ] [] [systemYamlHelper.sh:605       ] [main] - Resolved .artifactory.database.maxOpenConnections (80) from /opt/jfrog/artifactory/var/etc/system.yaml
2022-05-20T09:07:36.472Z [shell] [INFO ] [] [systemYamlHelper.sh:605       ] [main] - Resolved .access.database.maxOpenConnections (80) from /opt/jfrog/artifactory/var/etc/system.yaml
2022-05-20T09:07:36.859Z [shell] [INFO ] [] [systemYamlHelper.sh:605       ] [main] - Resolved .jfconnect.enabled (true) from /opt/jfrog/artifactory/var/etc/system.yaml
2022-05-20T09:07:37.039Z [shell] [INFO ] [] [installerCommon.sh:1597       ] [main] - Checking open files and processes limits
2022-05-20T09:07:37.056Z [shell] [INFO ] [] [installerCommon.sh:1600       ] [main] - Current max open files is 1048576
2022-05-20T09:07:37.076Z [shell] [INFO ] [] [installerCommon.sh:1611       ] [main] - Current max open processes is unlimited
Error: Failed to parse user system yaml: yaml: line 103: could not find expected ':'
Usage:
  diagnostics validate source-type source-file schema-file [flags]
Arguments
	source-type: json, yaml
	source-file: path to source file to validate
	schema-file: path to schema file

Flags:
  -h, --help                     help for validate
      --uncomment-before-parse   Do one level uncommenting on the source file before parsing
      --useJsonSchema            Use jsonschema to validate

2022-05-20T09:07:37.123Z [shell] [WARN ] [] [installerCommon.sh:799        ] [main] - System.yaml validation failed

Database connection check failed Could not determine database type
2022-05-20T09:07:37.231Z [shell] [INFO ] [] [installerCommon.sh:1680       ] [main] - Testing directory /opt/jfrog/artifactory/var has read/write permissions for user id 1030
2022-05-20T09:07:37.256Z [shell] [INFO ] [] [installerCommon.sh:1695       ] [main] - Permissions for /opt/jfrog/artifactory/var are good
```

**What you expected to happen**: Artifactory should start up without issue

**How to reproduce it** (as minimally and precisely as possible):
```yaml
# values.yaml (sensitive values redacted, of course)
artifactory:
  artifactory:
    customPersistentPodVolumeClaim:
      accessModes:
      - ReadWriteOnce
      mountPath: /backups
      name: artifactory-backups
      size: 50Gi
    customVolumeMounts: |
      - name: {{ .Values.artifactory.customPersistentPodVolumeClaim.name }}
        mountPath: {{ .Values.artifactory.customPersistentPodVolumeClaim.mountPath }}
    javaOpts:
      xms: 2g
      xmx: 3g
    livenessProbe:
      initialDelaySeconds: 600
    persistence:
      size: 50Gi
      type: file-system
    resources:
      limits:
        cpu: 2000m
        memory: 4Gi
      requests:
        cpu: 1m
        memory: 2Gi
  database:
    driver: org.postgresql.Driver
    password: xxxxxxxxxxxx
    type: postgresql
    url: jdbc:postgresql://xxxxxx.cluster-xxxxxxx.eu-west-1.rds.amazonaws.com:5432/artifactory
    user: xxxxxxxxxxxx
  fullnameOverride: artifactory-artifactory
  ingress:
    annotations:
      nginx.ingress.kubernetes.io/proxy-body-size: "0"
    className: nginx-internal
    defaultBackend:
      enabled: false
    enabled: true
    hosts:
    - artifactory.example.org
  nginx:
    enabled: false
    https:
      enabled: false
  postgresql:
    enabled: false
```

**Comment by rblaine95** [comment]
Yaml validation succeeds on downgrading back to `107.37.14`

**Comment by chukka** [comment]
@rblaine95  can you share `/opt/jfrog/artifactory/var/etc/system.yaml` file contents here

**Comment by rblaine95** [comment]
Hi @chukka, thank you for the response, please see `system.yaml` contents here:
```yaml
router:
    serviceRegistry:
        insecure: false
shared:
    logging:
        consoleLog:
            enabled: false
    extraJavaOpts: >
        -Dartifactory.access.client.max.connections=50 -Dartifactory.async.corePoolSize=8 -Xms2g -Xmx3g

    database:
        type: "postgresql"
        driver: "org.postgresql.Driver"
artifactory:
    database:
        maxOpenConnections: 80
    tomcat:
        maintenanceConnector:
            port: 8091
        connector:
            maxThreads: 200
            sendReasonPhrase: false
            extraConfig: acceptCount="100"
frontend:
    session:
        timeMinutes: "30"
access:
    database:
        maxOpenConnections: 80
    tomcat:
        connector:
            maxThreads: 50
            sendReasonPhrase: false
            extraConfig: acceptCount="100"
metadata:
    database:
        maxOpenConnections: 80
jfconnect:
    enabled: true
```

**Comment by rblaine95** [comment]
@chukka, apologies, I commented the wrong version of the `system.yaml` file, updated the comment above.

**Comment by rblaine95** [comment]
If I do a diff upgrade, I see this in the system yaml:
```diff
# helm diff upgrade -n dev-lifecycle-svc artifactory -f ./artifactory.yaml  jfrog/artifactory-oss --version 107.37.14 --show-secrets
dev-lifecycle-svc, artifactory-artifactory-systemyaml, Secret (v1) has changed:
  # Source: artifactory-oss/charts/artifactory/templates/artifactory-system-yaml.yaml
  apiVersion: v1
  kind: Secret
  metadata:
    name: artifactory-artifactory-systemyaml
    labels:
      app: artifactory
-     chart: artifactory-107.38.10
+     chart: artifactory-107.37.14
      heritage: Helm
      release: artifactory
  type: Opaque
  stringData:
    system.yaml: |
      router:
        serviceRegistry:
          insecure: false
      shared:
        logging:
          consoleLog:
            enabled: false
        extraJavaOpts: >
          -Dartifactory.access.client.max.connections=50
          -Dartifactory.async.corePoolSize=8
          -Xms2g
          -Xmx3g
        database:
          type: "postgresql"
          driver: "org.postgresql.Driver"
      artifactory:
        database:
          maxOpenConnections: 80
        tomcat:
          maintenanceConnector:
            port: 8091
          connector:
            maxThreads: 200
            sendReasonPhrase: false
            extraConfig: acceptCount="100"
      frontend:
        session:
          timeMinutes: "30"
      access:
        database:
          maxOpenConnections: 80
        tomcat:
          connector:
            maxThreads: 50
            sendReasonPhrase: false
            extraConfig: acceptCount="100"
      metadata:
        database:
          maxOpenConnections: 80
-     jfconnect:
-       enabled: true
```

**Comment by rblaine95** [comment]
I tested disabling jfconnect and the frontend successfully started up! 🥳 🎉 
```yaml
artifactory:
  jfconnect:
    enabled: false
```

Even so, the same error (`Error: Failed to parse user system yaml: yaml: line 103: could not find expected ':'`) still gets logged.
But the Artifactory frontend is now working. So it seems like it was an unrelated issue 🤔

**Comment by KarstenSiemer** [comment]
Hi! I have the exact same issue.
log:
```
artifactory-oss-0 artifactory Error: Failed to parse user system yaml: yaml: line 103: could not find expected ':'
artifactory-oss-0 artifactory Usage:
artifactory-oss-0 artifactory   diagnostics validate source-type source-file schema-file [flags]
artifactory-oss-0 artifactory Arguments
artifactory-oss-0 artifactory 	source-type: json, yaml
artifactory-oss-0 artifactory 	source-file: path to source file to validate
artifactory-oss-0 artifactory 	schema-file: path to schema file
artifactory-oss-0 artifactory
artifactory-oss-0 artifactory Flags:
artifactory-oss-0 artifactory   -h, --help                     help for validate
artifactory-oss-0 artifactory       --uncomment-before-parse   Do one level uncommenting on the source file before parsing
artifactory-oss-0 artifactory       --useJsonSchema            Use jsonschema to validate
artifactory-oss-0 artifactory
artifactory-oss-0 artifactory 2022-05-25T10:07:06.877Z [shell] [WARN ] [] [installerCommon.sh:799        ] [main] - System.yaml validation failed
```
helm values:
```
# Default values for artifactory-oss.
# This is a YAML-formatted file.

# Beware when changing values here. You should know what you are doing!
# Access the values with {{ .Values.key.subkey }}

# This chart is based on the main artifactory chart with some customizations.
# See all supported configuration keys in https://github.com/jfrog/charts/tree/master/stable/artifactory

## All values are under the 'artifactory' sub chart.
artifactory:

  ## Artifactory
  ## See full list of supported Artifactory options and documentation in artifactory chart: https://github.com/jfrog/charts/tree/master/stable/artifactory
  artifactory:
    ## Default tag is from the artifactory sub-chart in the requirements.yaml
    image:
      registry: releases-docker.jfrog.io
      repository: jfrog/artifactory-oss
      # tag:
    ## Uncomment the following resources definitions or pass them from command line
    ## to control the cpu and memory resources allocated by the Kubernetes cluster
    resources:
      requests:
        memory: "1Gi"
        cpu: "800m"
      limits:
        memory: "4Gi"
    ## The following Java options are passed to the java process running Artifactory.
    ## You should set them according to the resources set above.
    ## IMPORTANT: Make sure resources.limits.memory is at least 1G more than Xmx.
    javaOpts: {}
     # xms: "1g"
     # xmx: "3g"
     # other: ""

  installer:
    platform: art-oss-helm

  installerInfo: '{"productId": "Helm_artifactory-oss/{{ .Chart.Version }}", "features": [ { "featureId": "Platform/{{ default "kubernetes" .Values.installer.platform }}"}]}'

  ## Nginx
  ## See full list of supported Nginx options and documentation in artifactory chart: https://github.com/jfrog/charts/tree/master/stable/artifactory
  nginx:
    enabled: false
    tlsSecretName: ""
    service:
      type: LoadBalancer

  postgresql:
    enabled: true
    global:
      storageClass: gp2
      postgresql:
        auth:
          username: artifactory
          database: artifactory

  ## Ingress
  ## See full list of supported Ingress options and documentation in artifactory chart: https://github.com/jfrog/charts/tree/master/stable/artifactory
  ingress:
    enabled: true
    hosts:
      - artifactory-oss.xxx.xxx
    annotations:
      kubernetes.io/tls-acme: "true"
    ## Artifactory Ingress TLS configuration
    ## Secrets must be manually created in the namespace
    ##
    tls:
      - hosts:
        - artifactory-oss.xxx.xxx

  databaseUpgradeReady: "yes"

```

k8s secret:
```
router:
  serviceRegistry:
    insecure: false
shared:
  logging:
    consoleLog:
      enabled: false
  extraJavaOpts: >
    -Dartifactory.access.client.max.connections=50
    -Dartifactory.async.corePoolSize=8
  database:
    type: postgresql
    url: "jdbc:postgresql://artifactory-oss-postgresql:5432/artifactory"
    driver: org.postgresql.Driver
    username: "artifactory"
artifactory:
  database:
    maxOpenConnections: 80
  tomcat:
    maintenanceConnector:
      port: 8091
    connector:
      maxThreads: 200
      sendReasonPhrase: false
      extraConfig: acceptCount="100"
frontend:
  session:
    timeMinutes: "30"
access:
  database:
    maxOpenConnections: 80
  tomcat:
    connector:
      maxThreads: 50
      sendReasonPhrase: false
      extraConfig: acceptCount="100"
metadata:
  database:
    maxOpenConnections: 80
jfconnect:
  enabled: true
```

helm chart version:
artifactory-oss:  107.38.10

**Comment by srinivasgowda097** [comment]
This is already fixed internally and will be released along with Artifactory 7.41.x - ETA - End of this month

**Comment by chukka** [comment]
Fixed as part of https://github.com/jfrog/charts/pull/1651
